pkg/util/server/error_test.go

```diff
@@ -9,13 +9,12 @@ import (
 	"net/http/httptest"
 	"testing"
 
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
 	"github.com/grafana/dskit/httpgrpc"
 	"github.com/grafana/dskit/user"
 	"github.com/prometheus/prometheus/promql"
 	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
```

Comment:
Do you really want this dependency? The rest of the Loki repository uses `github.com/gogo/status`.
I also don't see where it is actually used in this source.

Reply:
This is older code. I think the assumption was that gRPC might return a status for something like `status.New(codes.Canceled, context.Canceled.Error())`. That's how it's used in the test cases. Would a gogo protobuf generated client return a `gogo.status.Status`?

Reply:
You can simply use `github.com/gogo/status`.
I've check the rest of the code. The `status` package is used only for creating errors. This is an example of its usage:
```
status.New(codes.Canceled, context.Canceled.Error()).Err()
```
This can be simplified by using:
```
status.Error(codes.Canceled, context.Canceled.Error())
```